portafly/src/components/data-list/modals/DataListModal.tsx

```diff
@@ -7,31 +7,44 @@ import {
   TextContent,
   TextList,
   Title,
-  BaseSizes
+  BaseSizes,
+  Alert
 } from '@patternfly/react-core'
 import { useTranslation } from 'i18n/useTranslation'
+import { useDataListBulkActions } from 'components/data-list'
 
 import './dataListModal.scss'
 
 interface Props {
-  onClose: () => void
   items: string[]
   title: string
-  actions: any
+  submitButton: JSX.Element
   to: string
+  errorMsg?: string
+  shouldWarnClose?: boolean
 }
 
 const DataListModal: React.FunctionComponent<Props> = ({
   children,
-  onClose,
   items,
   title,
-  actions,
-  to
+  submitButton,
+  to,
+  errorMsg,
+  shouldWarnClose
 }) => {
   const { t } = useTranslation('shared')
+  const { closeModal } = useDataListBulkActions()
+
   const [isListCollapsed, setIsListCollapsed] = useState(items.length > 5)
 
+  const onClose = () => {
+    // eslint-disable-next-line no-alert
+    if (!shouldWarnClose || window.confirm(t('modals.close_confirmation'))) {
+      closeModal()
+    }
+  }
+
```

Comment:
💄 
```suggestion
    // eslint-disable-next-line no-alert
    const onClose = () => if (!shouldWarnClose || window.confirm(t('modals.close_confirmation'))) closeModal()
```

Reply:
I am very respectfully ignoring this suggestion 😬